### ISSUE TITLE
Mouse-over element highlighting freezes

### ISSUE THREAD

**Issue by revcor** [Anze/KeyCluCask#33]
Normal behavior: When the main window is called up, moving the mouse cursor over the different shortcuts listed will cause each item to be highlighted while the cursor is hovering over it. 

Wonky behavior: If, **when** the main window is called up, the cursor is already at a point on the screen such that the KeyClu window opening causes one of the shortcut list items to appear **under** the cursor arrow, then that item will be stuck permanently highlighted, even if the mouse cursor is moved away from that item or onto other items. This resets when the main window is dismissed.

<img width="1291" alt="Screen Shot 2023-09-26 at 1 28 46 p m" src="https://github.com/Anze/KeyCluCask/assets/94559820/82673c74-c6aa-4cfb-840b-b49e20ec4685">
<img width="1289" alt="Screen Shot 2023-09-26 at 1 28 59 p m" src="https://github.com/Anze/KeyCluCask/assets/94559820/25deb099-36ce-4722-81de-8b3370fab57e">

**Comment by Anze** [comment]
Hey [revcor](https://github.com/revcor), thanks for reporting the issue. Could you please let me know which macOS version you're currently running?
I've come across another issue, the grid should always begin at the top. This is something I haven't encountered on Big Sur, Ventura, or Sonoma before.

**Comment by revcor** [comment]
Woops that's probably important haha. Intel mac running Monterey 12.7 (Mid 2015 MBP dual GPU Retina 15-inch in case it matters).

I had also noticed the empty space above some of the columns. In the app preferences I fiddled with the different values for "Columns" and toggling "Show inactive elements" and nothing had any effect on the empty spaces.

**Comment by Anze** [comment]
Hey [revcor](https://github.com/revcor), I've made some fixes for macOS 12, could you please provide a bit of assistance in verifying whether these fixes have made a difference.

**Comment by revcor** [comment]
Absolutely, anything I can do to help. I can't get to it right at this moment but in a few hours I'll be free

**Comment by revcor** [comment]
https://github.com/Anze/KeyCluCask/assets/94559820/073e1f6c-a240-412c-91d4-f411165a40fa

**v0.14 initial testing**

I did all the actions in the video quickly to keep video file size down, so you may want to pause it at various points to see the relevant parts

**Observations:**
- The columns are all flush with the top of the window now
- In some cases, not all columns are populated, leaving some stacked from the left, then empty space, and remainder on the far right (only seen with Finder selected as the active app)
- Single item highlighting issue persists: If window called up on top of the pointer location, and a shortcut list item occupies the same space as pointer, that item will be permanently highlighted even after moving pointer away.  If window called up does _not_ overlap pointer, then moving pointer over list items does not cause **any** items to be highlighted

Apologies for delay in getting this to you

**Comment by Anze** [comment]
Thanks for the report.
It's great to hear that it's already making a difference, especially in terms of the visual aspect. Still, there are some issues that can't be resolved immediately, as they require a more in-depth investigation. One of these issues is the substantial memory leak in macOS 11 & 12, which is not present in macOS 13 & 14. I'm working on performance improvements to mitigate these memory leaks, although they haven't been completely eliminated yet.
As for the empty columns, I've made improvements of the distribution and fill columns more evenly. While it's not a universal solution for all cases, it should still provide a better overall experience.

**Comment by revcor** [comment]
Definitely an improvement! Any way I can be of assistance, please let me know. Under what conditions does this memory leak happen?

**Comment by Anze** [comment]
One of the case happens when elastic scrolling reaches the top or bottom.

**Comment by Anze** [comment]
Hey [revcor](https://github.com/revcor),
I've just released a new version. It would be greatly appreciated if you could check to see if there are any differences.

**Comment by revcor** [comment]
Grabbing it now

**Comment by Anze** [comment]
Hey [revcor](https://github.com/revcor),
I've just released a new version with significant performance enhancements. Would love to hear your thoughts on it.
